### ISSUE TITLE
`Series.[q]cut`: available as Expression, accepts int parameter, `break_point` column as optional

### ISSUE THREAD

**Issue by baggiponte** [pola-rs/polars#9468]
### Problem description

# Proposals
1. When `Series.qcut`'s `quantile` param is an integer, it is interpreted as the number of quantiles (e.g. 4 -> [0.0, 0.25, 0.5, 0.75, 1]).
2. A new parameter `return_breakpoint` that can suppress returning the `break_point` column (though this complicates things quite a bit - see below)
3. `qcut` as an expression: `Expr.qcut()`

This would make `qcut` closer to what `pandas.qcut` does (though not even pyspark supports this use case)

# Cons

Suppose I have this data:

```python
data = pl.DataFrame({
    "users": ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"],
    "class": ["a", "a", "a", "b", "b", "b", "c", "c", "c", "c"],
    "scores": [12, 34, 56, 78, 90, 12, 34, 56, 78, 90],
})
```

Right now I can only run `data["scores"].qcut([0.1, 0.5, 1])` - so I cannot use it inside any context (groupby, window function...). Plus I need a join to put in back with the original data. If I needed this bit of info, right now I have to turn the data into a `pandas.DataFrame`:

```python
data.to_pandas().assign(quartile = lambda df: pd.qcut(df["scores"], 4))
```

With the suggested proposal, I could write:

```python
data.with_columns(pl.col("scores").qcut(3, labels=[str(x) for x in range(3 + 1)])
```

Note:  the `labels` params is a bit weird, as right now it does not accept a list of int.

This seems nice, but there are a bunch of problems:

1. What is the return type if the break point column is not returned?
2. If we *do* return the break point, should it return a `Struct` with two fields, `break_point_label` and `category_label`?
3. The parameters to decide the name of the columns are confusing. What syntax should we support?

```python
data.with_columns(pl.col("scores").qcut(3, labels=[str(x) for x in range(3 + 1)], return_breakpoint=False, category_label="quartile").struct.field("quartile")

data.with_columns(pl.col("scores").qcut(3, labels=[str(x) for x in range(3 + 1)], return_breakpoint=False).alias("quartile")
```

Besides, making `qcut` an expression would require rewriting the internals of `Series.hist`.

# Alternatives

Alternatively, there could be a `DataFrame.qcut` method that accepts a `column` parameter as input and does the `qcut` magic for that column only, though we still retain the problem of the return type.

# Contribution

I could try/would be glad to work on this, but I don't know rust (though I think I can manage to create a vector of integers (^^"))

**Comment by baggiponte** [comment]
Basically closed by #9745.

> 1. When Series.qcut's quantile param is an integer, it is interpreted as the number of quantiles (e.g. 4 -> [0.0, 0.25, 0.5, 0.75, 1]).
> 2. A new parameter return_breakpoint that can suppress returning the break_point column (though this complicates things quite a bit - see below)

@ritchie46 what do you think about these? If you do not feel like polars should support this, I will proceed to close the issue!

**Comment by magarick** [comment]
The new defaults will just return a series of categories.
Evenly spaced quantiles is a good feature to have, I just haven't gotten around to it yet. I think it should be a separate function though, maybe call it `nqcut`. It continues to amaze me how many details matter for such a seemingly simple operation.

**Comment by baggiponte** [comment]
I went with the same function just because pandas does so and it might be more familiar, but of course it makes sense to create a different function.

Also I'd scrap 3 since you can already return just the labels if you pass them, or a struct with labels and thresholds.

**Comment by baggiponte** [comment]
I guess this is closed by #9960!
